Source/AliveLib/Input.cpp

```diff
@@ -910,82 +932,54 @@ void NewParseSettingsIni()
-                        {
-                            sJoystickEnabled_5C9F70 = 1;
-                        }
-                    }
-                    else if (param[0] == "buttons")
-                    {
-                        if (sJoystickNumButtons_5C2EFC && atol(param[1].c_str()) != sJoystickNumButtons_5C2EFC)
-                        {
-                            isKeyboardParams = false;
-                        }
-                        else
-                        {
-                            Input_ResetBinding_4925A0(16, 1);
-                            Input_ResetBinding_4925A0(64, 1);
-                            Input_ResetBinding_4925A0(256, 1);
-                            Input_ResetBinding_4925A0(32, 1);
-                            Input_ResetBinding_4925A0(128, 1);
-                            Input_ResetBinding_4925A0(512, 1);
-                            Input_ResetBinding_4925A0(0x800000, 1);
-                            Input_ResetBinding_4925A0(InputCommands::eSpeak2, 1);
-                        }
-                    }
-                    else
+                    InputCommands gamepadInputCommand = Input_LoadSettingsIni_GetInputCommand_492B80(param[0].c_str());
+                    Input_SetGamePadBinding_4931D0(param[1].c_str(), gamepadInputCommand);
+                }
+                else if (currentCategory == IniCategory::eAlive)
+                {
+                    for (const IniCustomSaveEntry& s : gCustomSaveEntries)
                     {
-                        bool foundConfig = false;
-                        // custom config loader code
+                        if (param[0] == s.name)
                         {
-                            for (const IniCustomSaveEntry& s : gCustomSaveEntries)
+                            if (s.isBool)
                             {
-                                if (param[0] == s.name)
+                                if (param[1] == "true")
                                 {
-                                    if (s.isBool)
-                                    {
-                                        if (param[1] == "true")
-                                        {
-                                            *reinterpret_cast<bool*>(s.data) = true;
-                                        }
-                                        else
-                                        {
-                                            *reinterpret_cast<bool*>(s.data) = false;
-                                        }
-                                    }
-                                    else // int
-                                    {
-                                        *reinterpret_cast<int*>(s.data) = atoi(param[1].c_str());
-                                    }
-
-                                    foundConfig = true;
-                                    break;
+                                    *reinterpret_cast<bool*>(s.data) = true;
```

Comment:
Existing but change the structure to just have an int and then add an AsInt() AsBool() method or something, or make it a union, anything but this :)